extensions/ql-vscode/src/query-results.ts

```diff
@@ -160,10 +161,12 @@ export async function interpretResultsSarif(
   sourceInfo?: cli.SourceInfo
 ): Promise<SarifInterpretationData> {
   const { resultsPath, interpretedResultsPath } = resultsPaths;
+  let res;
   if (await fs.pathExists(interpretedResultsPath)) {
-    return { ...JSON.parse(await fs.readFile(interpretedResultsPath, 'utf8')), t: 'SarifInterpretationData' };
+    res = await sarifParser(interpretedResultsPath);
```

Comment:
I think you still need the extra `t: 'SarifInterpretationData'` property, so the results view knows what to do with this object. See what we do just below.

Reply:
I am appending it a few lines down in the `return` statement — does it need to be assigned to `res` here?

Reply:
Gah didn't read the diff closely enough. You have done exactly that!